samples/packages/asset-import/src/test/kotlin/com/atlan/pkg/aim/ImportDataDomainTest.kt

```diff
@@ -0,0 +1,186 @@
+import org.testng.annotations.BeforeClass
+import java.nio.file.Paths
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+
+class ImportDataDomainTest : PackageTest() {
+
+    private val dataDomain1 = makeUnique("idd1")
+    private val dataDomain2 = makeUnique("idd2")
+    private val dataDomain3 = makeUnique("idd3")
+    private val dataProduct1 = makeUnique("idp1")
+    private val tag1 = makeUnique("idt1")
+
+    private val testFile = "input.csv"
+
+    private val files = listOf(
+        testFile,
+        "debug.log",
+    )
+
+    private fun prepFile() {
+        // Prepare a copy of the file with unique names for glossaries and tags
+        val input = Paths.get("src", "test", "resources", "data_domain.csv").toFile()
+        val output = Paths.get(testDirectory, testFile).toFile()
+        input.useLines { lines ->
+            lines.forEach { line ->
+                val revised = line
+                    .replace("{{DATADOMAIN1}}", dataDomain1)
+                    .replace("{{DATADOMAIN2}}", dataDomain2)
+                    .replace("{{DATADOMAIN3}}", dataDomain3)
+                    .replace("{{DATAPRODUCT1}}", dataProduct1)
+                    .replace("{{TAG1}}", tag1)
+                output.appendText("$revised\n")
+            }
+        }
+    }
+
+    private fun createTags() {
+        val maxNetworkRetries = 30
+        val client = Atlan.getDefaultClient()
+        val t1 = AtlanTagDef.creator(tag1, AtlanIcon.AIRPLANE, AtlanTagColor.GREEN).build()
+//        val t2 = AtlanTagDef.creator(tag2, AtlanIcon.ROBOT, AtlanTagColor.RED).build()
+        client.typeDefs.create(
+            listOf(t1),
+            RequestOptions.from(client).maxNetworkRetries(maxNetworkRetries).build(),
+        )
+    }
+
+    private val dataDomainAttrs: List<AtlanField> = listOf(
+        DataDomain.NAME,
+        DataDomain.ASSET_ICON,
+        DataDomain.ASSET_THEME_HEX,
+        DataDomain.ASSET_COVER_IMAGE,
+        DataDomain.USER_DESCRIPTION,
+        DataDomain.OWNER_USERS,
+        DataDomain.OWNER_GROUPS,
+        DataDomain.CERTIFICATE_STATUS,
+        DataDomain.CERTIFICATE_STATUS_MESSAGE,
+        DataDomain.PARENT_DOMAIN,
+        DataDomain.PARENT_DOMAIN_QUALIFIED_NAME,
+        DataDomain.SUPER_DOMAIN_QUALIFIED_NAME,
+    )
+
+    private val dataProductAttrs: List<AtlanField> = listOf(
+        DataProduct.NAME,
+        DataProduct.ASSET_COVER_IMAGE,
+        DataProduct.USER_DESCRIPTION,
+        DataProduct.OWNER_USERS,
+        DataProduct.OWNER_GROUPS,
+        DataProduct.CERTIFICATE_STATUS,
+        DataProduct.CERTIFICATE_STATUS_MESSAGE,
+        DataProduct.DATA_DOMAIN,
+    )
+
+    @BeforeClass
+    fun beforeClass() {
+        prepFile()
+//        createTags()
+        setup(
+            AssetImportCfg(
+                assetsFile = null,
+                assetsUpsertSemantic = null,
+                assetsAttrToOverwrite = null,
```

Comment:
Any config options you don't specify should already be set to `null` by default, so might be easiest to just leave them out entirely here.

Reply:
Changed.